packages/android_intent/lib/android_intent.dart

```diff
@@ -36,7 +36,7 @@ class AndroidIntent {
     this.arguments,
     this.package,
     this.componentName,
-    Platform platform,
+    Platform? platform,
     this.type,
   })  : assert(action != null || componentName != null,
```

Comment:
could you remind me why `assert` is still needed in g3?

Reply:
This particular assert is useful because it checks that either `action` or `componentName` is non-null. Therefore, both are nullable.

For non-nullable vars, asserts are useful for mixed mode. If an app that is not yet migrated uses a null-safe library, they can still pass null to a non-nullable type. This will eventually fail with a _CastError if we remove the asserts. The assert is more explicit and meaningful. When most code in Dart ecosystem is migrated to null-safety and/or we have had a couple of stable releases of nnbd, we can start cleaning up the asserts. package:flutter is following the same thing.

Reply:
Makes sense! thanks